docs/source/tutorials_source/package/tutorial_custom_augmentations.py

```diff
@@ -0,0 +1,366 @@
+example_image_path = os.path.join(path_to_data, example_image_name)
+example_image = np.array(Image.open(example_image_path))
+
+# torch transform returns a 3 x W x H image, we only show one color channel
+augmented_image_1 = transform(example_image).numpy()[0]
+augmented_image_2 = transform(example_image).numpy()[0]
+
+fig, axs = plt.subplots(1, 3)
+
+axs[0].imshow(example_image)
+axs[0].set_axis_off()
+axs[0].set_title('Original Image')
+
+axs[1].imshow(augmented_image_1)
+axs[1].set_axis_off()
+
+axs[2].imshow(augmented_image_2)
+axs[2].set_axis_off()
+
+# %%
+# Finally, in order to use the augmentation pipeline we defined for self-supervised
+# learning, we need to create a Lightly collate function like so:
+
+# create a collate function which performs the random augmentations
+collate_fn = lightly.data.BaseCollateFunction(transform)
+
+# %%
+# Setup dataset and dataloader
+# ------------------------------
+#
+# We create a dataset which points to the images in the input directory. Since
+# the input images are 16 bits deep, we need to overwrite the image loader such 
+# that it doesn't convert the images to RGB (and hence to 8-bit) automatically.
+
+
+def tiff_loader(f):
+    """Loads a 16-bit tiff image and returns it as a numpy array.
+
+    """
+    with open(f, 'rb') as f:
+        image = Image.open(f)
+        return np.array(image)
+
+# create the dataset and overwrite the image loader
+dataset_train = lightly.data.LightlyDataset(input_dir=path_to_data)
+dataset_train.dataset.loader = tiff_loader
+
+# setup the dataloader for training, make sure to pass the collate function
+# from above as an argument
+dataloader_train = torch.utils.data.DataLoader(
+    dataset_train,
+    batch_size=batch_size,
+    shuffle=True,
+    collate_fn=collate_fn,
+    drop_last=True,
+    num_workers=num_workers
+)
+
+# %%
+# Create the MoCo model
+# -----------------------
+# We can chop off the linear classification head of a ResNet to get a backbone 
+# neural network.
+
+# remove the classification head and get the ResNet backbone
+resnet = lightly.models.ResNetGenerator('resnet-18')
+backbone = nn.Sequential(
+    *list(resnet.children())[:-1],
+    nn.AdaptiveAvgPool2d(1)
+)
+
+# %%
+# Now, we can use the backbone to create the MoCo model. It's important, that
+# the number of features matches the output dimension of the backbone.
+model = lightly.models.MoCo(backbone, num_ftrs=512, m=0.99)
+
+# %%
+# Setup criterion and optimizer
+# -----------------------------
+# For the criterion, we use the NTXentLoss which should always be used with MoCo.
+# MoCo also requires a memory bank - we set its size to 4096.
+#
+# The choice of the optimizer is left to the user. Here, we go with simple stochastic
+# gradient descent with momentum.
+criterion = lightly.loss.NTXentLoss(
+    temperature=0.1,
+    memory_bank_size=4096,
+)
+
+optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9, weight_decay=1e-4)
```

Comment:
Same here, why do we use this lr? momentum is usually 0.9 (that's fine for me). But also weight_decay. Number from paper or "this just worked best for us".